lib/vagrant-service-manager/binary_handlers/docker_binary_handler.rb

```diff
@@ -0,0 +1,60 @@
+module VagrantPlugins
+  module ServiceManager
+    class DockerBinaryHandler
+      attr_reader :binary_name, :version
+      # Refer https://docs.docker.com/v1.10/engine/installation/binaries
+      DOCKER_BINARY_BASE_URL = 'https://get.docker.com/builds'
+
+      def initialize(machine, env, options)
+        @machine = machine
+        @version = options['--cli-version']
+        @type = options[:type]
+        @binary_name = BINARY_NAME[@type] + binary_ext
```

Comment:
Is the binary name not really a given? Why the need for a constant defined somewhere else?